app/assets/javascripts/admin/addon/models/color-scheme.js

```diff
@@ -76,6 +60,10 @@ export default class ColorScheme extends EmberObject {
   init() {
     super.init(...arguments);
 
+    const colors = A(this.colors ?? []);
+    this.colors = colors.map((c) => {
```

Comment:
All these looping in the PR is also making me wonder if we selected the wrong data structure here. Perhaps we should be using a object here instead of an array so that we can easily look up a color object by its name?

Reply:
I've eliminated all the loops except for this one. I think we'd still need to loop here even if we changed the data structure because this converts the data from POJOs to `ColorSchemeColor` objects.